Sources/SwiftLanguageService/CodeCompletionSession.swift

```diff
@@ -704,6 +714,50 @@ class CodeCompletionSession {
 
     return Position(line: completionPos.line, utf16index: deletionStartUtf16Offset)
   }
+
+  /// Generate a snippet for control flow keywords like if, for, while, etc.
+  /// Returns the snippet text if the keyword is a control flow keyword and snippets are supported, otherwise nil.
+  private func keywordSnippet(for keyword: String) -> String? {
+    guard clientSupportsSnippets else { return nil }
+
+    func indentationUnitString() -> String {
+      // Convert the inferred Trivia (spaces/tabs) to a string we can embed into snippets.
+      if let trivia = indentationWidth {
+        for piece in trivia {
+          switch piece {
+          case .spaces(let n):
+            return String(repeating: " ", count: n)
+          case .tabs(let n):
+            return String(repeating: "\t", count: n)
+          default:
+            continue
+          }
+        }
+      }
+      // Fallback to a single tab to preserve previous behaviour when indentation cannot be inferred.
+      return "\t"
+    }
+
+    let indent = indentationUnitString()
+    let doubleIndent = indent + indent
+
+    switch keyword {
+    case "if":
+      return "if ${1:condition} {\n\(indent)${0:}\n}"
+    case "for":
+      return "for ${1:item} in ${2:sequence} {\n\(indent)${0:}\n}"
+    case "while":
+      return "while ${1:condition} {\n\(indent)${0:}\n}"
+    case "guard":
+      return "guard ${1:condition} else {\n\(indent)${0:}\n}"
+    case "switch":
+      return "switch ${1:value} {\n\(indent)case ${2:pattern}:\n\(doubleIndent)${0:}\n}"
```

Comment:
Let’s not indent the `case` label to match the default behavior of [swift-format](https://github.com/swiftlang/swift-format/blob/main/Documentation/Configuration.md#indentswitchcaselabels)

Ie. make this

```suggestion
      return """
        switch ${1:value} {
        case ${2:pattern}:
        \(indent)${0:}
        }
        """
```